src/Commands/CommercePrintfulCommands.php

```diff
@@ -0,0 +1,178 @@
+
+use Consolidation\AnnotatedCommand\AnnotationData;
+use Drupal\commerce_printful\Service\Printful;
+use Drupal\commerce_printful\Service\ProductIntegrator;
+use Drupal\commerce_printful\PrintfulSyncBatch;
+use Drupal\commerce_printful\Exception\PrintfulException;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drush\Commands\DrushCommands;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+
+/**
+ * A Drush commandfile.
+ *
+ * In addition to this file, you need a drush.services.yml
+ * in root of your module, and a composer.json file that provides the name
+ * of the services file to use.
+ *
+ * See these files for an example of injecting Drupal services:
+ *   - http://cgit.drupalcode.org/devel/tree/src/Commands/DevelCommands.php
+ *   - http://cgit.drupalcode.org/devel/tree/drush.services.yml
+ */
+class CommercePrintfulCommands extends DrushCommands {
+
+  /**
+   * Drupal\commerce_printful\Service\Printful definition.
+   *
+   * @var \Drupal\commerce_printful\Service\Printful
+   */
+  protected $pf;
+
+  /**
+   * Drupal\commerce_printful\Service\ProductIntegrator definition.
+   *
+   * @var \Drupal\commerce_printful\Service\ProductIntegrator
+   */
+  protected $integrator;
+
+  /**
+   * Drupal\Core\Entity\EntityTypeBundleInfo definition.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeBundleInfo
+   */
+  protected $bundleInfo;
+
+  /**
+   * Drupal\Core\Config\ConfigFactoryInterface definition.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $printfulConfig;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    Printful $printful,
+    ProductIntegrator $product_integrator,
+    EntityTypeBundleInfoInterface $bundle_info,
+    ConfigFactoryInterface $config_factory
+  ) {
+    $this->pf = $printful;
+    $this->integrator = $product_integrator;
+    $this->bundleInfo = $bundle_info;
+    $this->printfulConfig = $config_factory;
+  }
+
+
+    /**
+     * @return \Drupal\Core\Config\ConfigFactoryInterface
+     */
+  public function getConfig() {
+      return $this->printfulConfig;
+  }
+
+  /**
+   * Test the Printful API connection.
+   *
+   *
+   * @command printful:test
+   * @aliases pt,printful-test
+   */
+  public function test() {
+    try {
+      $store_data = $this->pf->getStoreInfo();
+      if ($store_data['code'] == 200 && isset($store_data['result'])) {
+        $d = print_r($store_data['result'], TRUE);
+        $this->output()->writeln(dt('Store Info:'));
+        $this->output()->writeln($d);
+      }
+      else {
+        $this->output()->writeln(dt('No Store Info found.'));
+      }
+    }
+    catch (PrintfulException $e) {
+      dt($e->getMessage);
+    }
+    try {
+      $products = $this->pf->syncProducts();
+      if ($products['code'] == 200 && isset($products['result'])) {
```

Comment:
It's not going to work as the Printful service throws an exception in such a case. Just do a try - catch block and catch a PrintfulException.

Reply:
Ah, that makes sense.